tools/python_tools/cuvslam_tools/tests/test_icl_nuim_conversion.py

```
@@ -0,0 +1,401 @@
+            convert_icl_nuim.read_associations("1 depth/1.png 1\n", "assoc")
+
+    def test_mismatched_keys_are_rejected(self):
+        with self.assertRaisesRegex(convert_icl_nuim.ConversionError, "does not match"):
+            convert_icl_nuim.read_associations("1 depth/1.png 2 rgb/2.png\n", "assoc")
+
+    def test_duplicate_frame_is_rejected(self):
+        with self.assertRaisesRegex(convert_icl_nuim.ConversionError, "duplicate frame"):
+            convert_icl_nuim.read_associations(_associations_text([1, 1]), "assoc")
+
+    def test_unexpected_media_path_is_rejected(self):
+        with self.assertRaisesRegex(convert_icl_nuim.ConversionError, "unexpected media path"):
```

Ruff (0.16.3)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)